src/main/java/io/nats/client/impl/NatsDispatcher.java

```diff
@@ -106,6 +106,11 @@ public void run() {
                                 handler.onMessage(msg);
                             } catch (Exception exp) {
                                 connection.processException(exp);
+                            } catch (Error err) {
+                                // Make sure we log, so we don't silently exit our loop.
+                                // https://github.com/nats-io/nats.java/issues/1198
+                                err.printStackTrace();
```

Comment:
what about
```
connection.processException(new Exception(err));
```

Reply:
I'm leaning more toward logging and failing, than still processing the error and continuing on.
I'm thinking an `Error` indicates something that's not recoverable, versus an `Exception` being something that's just unexpected or can be retried.

In the case that the `Error` is not recoverable, there is probably not much we can do to keep running anyway. Might as well log and quit.

Reply:
Changed to indeed process as exception and continue.